src/com/google/javascript/jscomp/deps/DependencyInfo.java

```diff
@@ -140,6 +140,13 @@ abstract static class Builder {
   /** Whether the symbol is provided by a module */
```

Comment:
Should `isModule()` and `isGoogModule()` always return the same thing?

If so let's be more explicit about that - could we make DependencyInfo.Base.isModule `final`? The only overrides I see are in JSChunk.java and DependencyMap.java.

Reply:
Probably, but I'm not totally sure that it is always the same thing. From JSChunk.isModule():
```
  @Override
  public boolean isModule() {
    // NOTE: The meaning of "module" has changed over time.  A "JsChunk" is
    // a collection of inputs that are loaded together. A "module" file,
    // is a CommonJs module, ES6 module, goog.module or other file whose
    // top level symbols are not in global scope.
    throw new UnsupportedOperationException();
  }
```
I was concerned that, despite the implementation in DependencyInfo.Base.isModule() explicitly checking for goog modules, that this JsChunk implementation required that isModule continue to throw. The comment _seems_ to be generally inaccurate, or at least in conflict with other implementations, so I only changed DependencyInfo.Base.isModule() to take the fast path (and skip parsing), but stil permit JSChunk to throw in this specific case.

I didn't notice at the time (but am seeing now) that JSChunk also throws for getLoadFlags(), so perhaps I'm being overly cautious. Will be guided by your knowledge/preference here - if I'm being too careful about code that I can't see (that assumes isModule() might in fact be referring to commonjs/es6/goog as the comment describes), yes, we can probably delete JSChunk.isModule().

--

I can see no source file with the name DependencyMap.java, nor type with that name - typo, or internal class I can't see?